card/src/main/java/com/adyen/checkout/card/NewCardDelegate.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2020 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by caiof on 13/11/2020.
+ */
+
+package com.adyen.checkout.card
+
+import com.adyen.checkout.card.data.CardType
+import com.adyen.checkout.card.data.ExpiryDate
+import com.adyen.checkout.components.model.paymentmethods.PaymentMethod
+import com.adyen.checkout.components.util.PaymentMethodTypes
+import com.adyen.checkout.components.validation.ValidatedField
+
+class NewCardDelegate(private val paymentMethod: PaymentMethod) : CardDelegate {
+    override fun getPaymentMethodType(): String {
+        return paymentMethod.type ?: PaymentMethodTypes.UNKNOWN
+    }
+
+    override fun validateCardNumber(cardNumber: String): ValidatedField<String> {
+        return CardValidationUtils.validateCardNumber(cardNumber)
+    }
+
+    override fun validateExpiryDate(expiryDate: ExpiryDate): ValidatedField<ExpiryDate> {
+        return CardValidationUtils.validateExpiryDate(expiryDate)
+    }
+
+    override fun validateSecurityCode(
+        securityCode: String,
+        cardConfiguration: CardConfiguration,
+        cardType: CardType?
+    ): ValidatedField<String> {
+        return if (cardConfiguration.isHideCvc) {
+            ValidatedField(securityCode, ValidatedField.Validation.VALID)
+        } else {
+            CardValidationUtils.validateSecurityCode(securityCode, cardType)
+        }
+    }
+
+    override fun isCvcHidden(cardConfiguration: CardConfiguration): Boolean {
```

Comment:
@jreij do you think we should add the `CardConfiguration` to the constructor instead?

Reply:
Hmm yeah why not